http_logs/challenges/default.json

```diff
@@ -23,11 +23,22 @@
             }
           }
         },
-        {
-          "operation": "index-append",
-          "warmup-time-period": 240,
-          "clients": {{bulk_indexing_clients | default(8)}}
-        },
+        {%- if runtime_script_grok is defined %}
```

Comment:
@nik9000 In order to visualize the queries in https://elasticsearch-benchmarks.elastic.co/#tracks/http-logs/nightly/default/30d we will actually need a second challenge. The query visualizations and the chart generator will not current visualize more than one operations in the same chart (plus it'd get very noisy given that they already visualize 4 percentiles).

So I suggest we create a new challenge e.g. `append-no-conflicts-with-runtime-fields` or something like that, where we have all the content of the `append-no-conflicts` challenge plus the operations in the conditionals here. We can skip the `if` in the new challenge.

Reply:
I have this condition that I call copy-and-paste blindness. If I see a bunch of stuff that looks copy and pasted I can never find the small bits that differ. Is there any way I can write this as a function and call it twice to get two named challenges? That way the if statement stay? Otherwise I'll never be able to find the difference.

Reply:
I hear you. You can leverage the `{{ rally.collect(parts="") }}` helper to reference common sections.

Example approach:

1. Create a `common` dir under challenges, e.g. like:

```
├── challenges
│   ├── common
│   │   └── default-schedule.json
│   └── default.json
```

2. Move the the schedule from the `append-no-conflicts` challenge to `common/default-schedule.json`.
3. Change the `append-no-conflicts` challenge to be:

    ```
      {
        "name": "append-no-conflicts",
        "description": "Indexes the whole document corpus using Elasticsearch default settings. We only adjust the number of replicas as we benchmark a single node cluster and Rally will only start the benchmark if the cluster turns green. Document ids are unique so all index operations are append only. After that a couple of queries are run.",
        "default": true,
        "schedule": [
          {{ rally.collect(parts="common/default-schedule.json") }}
        ]
      },
    ```

4. Create a second challenge with a different name + description, same content, that relies on use specifying the `runtime_script_grok` track parameter.

This is just an idea, one could split sections from the schedule of the default challenge to more files and include as required.